src/sif/autocost.cc

```diff
@@ -455,7 +455,12 @@ Cost AutoCost::EdgeCost(const baldr::DirectedEdge* edge,
   // either the computed edge speed or optional top_speed
   auto edge_speed = tile->GetSpeed(edge, flow_mask_, seconds, false, &flow_sources);
   auto final_speed = std::min(edge_speed, top_speed_);
-  float sec = edge->length() * speedfactor_[final_speed];
+  // Use reduced speed for costing purposes, and normal edge speed for calculating duration (sec).
+  // This means closed edges cost more, but have the same traversal time as if they were open
+  auto speed_for_costing = IsClosed(edge, tile) ? kMinSpeedKph : final_speed;
+
+  float sec = (edge->length() * speedfactor_[final_speed]);
+  float cost_duration = (edge->length() * speedfactor_[speed_for_costing]);
```

Comment:
another approach we have taken here is to add a fixed penalty as a multiplier (factor) to any closed edge. rather than computing another speed, you just multiply it. and we've also made these request parameters so we can tune them if we need to rather than hardcoding a single penalty. another option is having a length of time penalty which is added to the final value and also configurable at request time. i personally think either of these options is a little cleaner than calculating a duration based on a sepcific speed but i dont feel strongly about it other than it being so different than the existing methods for penalization

Reply:
I've switched to using a factor thats multiplied to the final cost if traversing a closed edge. I derived the value comparing how high the cost was using the previous implementation of using reduced speed for costing.

CC: @danpat